airbyte-cdk/python/airbyte_cdk/connector.py

```diff
@@ -51,12 +60,25 @@ def write_config(config: Mapping[str, Any], config_path: str):
     def spec(self, logger: logging.Logger) -> ConnectorSpecification:
         """
         Returns the spec for this integration. The spec is a JSON-Schema object describing the required configurations (e.g: username and password)
-        required to run this integration.
+        required to run this integration. By default, this will be loaded from a "spec.yaml" or a "spec.json" in the package root.
         """
-        raw_spec: Optional[bytes] = pkgutil.get_data(self.__class__.__module__.split(".")[0], "spec.json")
-        if not raw_spec:
-            raise ValueError("Unable to find spec.json.")
-        return ConnectorSpecification.parse_obj(json.loads(raw_spec))
+
+        package = self.__class__.__module__.split(".")[0]
+
+        yaml_spec = load_optional_package_file(package, "spec.yaml")
+        json_spec = load_optional_package_file(package, "spec.json")
+
+        if yaml_spec and json_spec:
+            raise ValueError("Found multiple spec files in the package. Only one of spec.yaml or spec.json should be provided.")
```

Comment:
nit: this isn't really a [ValueError](https://docs.python.org/3.9/library/exceptions.html#ValueError), I'd go with the generic `RuntimeError`.